binzx/otomi

```diff
@@ -0,0 +1,307 @@
+      ;;
+    *)
+      return 0
+      ;;
+    esac
+  fi
+}
+silent echo "Checking for updates"
+[ -z $CI ] && check_update
+
+silent echo "Preparing docker environment variables"
+tmp_env=$(mktemp)
+
+function dump_vars() {
+  for var in "$@"; do
+    val="${!var}"
+    [ -n "$val" ] && echo "$var=$val" >> $tmp_env
+  done
+}
+
+OTOMI_PORT=${OTOMI_PORT:-192019}
+OTOMI_IN_TERMINAL="false"
+if [ -t 1 ]; then
+  # shellcheck disable=SC2034
+  OTOMI_IN_TERMINAL="true"
+fi
+
+vars=( 
+  AWS_ACCESS_KEY_ID 
+  AWS_REGION 
+  AWS_SECRET_ACCESS_KEY 
+  AZURE_CLIENT_ID 
+  AZURE_CLIENT_SECRET 
+  AZURE_TENANT_ID 
+  CI 
+  DEBUG
+  ENV_DIR
+  GCLOUD_SERVICE_KEY 
+  K8S_CONTEXT 
+  KUBE_VERSION_OVERRIDE 
+  OTOMI_DRY_RUN 
+  OTOMI_IN_TERMINAL
+  OTOMI_PASSWORD 
+  OTOMI_PORT 
+  OTOMI_SERVER 
+  OTOMI_TAG 
+  OTOMI_USERNAME 
+  PROFILE 
+  STATIC_COLORS 
+  TESTING 
+  TRACE 
+  VAULT_TOKEN 
+)
+dump_vars "${vars[@]}"
+
+cat >>$tmp_env <<EOF
+SHELL=${path_to_exec_shell}
+OTOMI_CALLER_COMMAND=${BASH_SOURCE[0]##*/}
+OTOMI_IN_DOCKER=true
+EOF
+
+helm_config="$HOME/.config/helm"
+if [[ $(uname -s) == "Darwin" ]]; then 
+  helm_config="$HOME/Library/Preferences/helm"
+else
+  readonly linux_workaround='--user=root:root'
+fi
+
+stack_dir='/home/app/stack'
+executable="node --experimental-specifier-resolution=node ${stack_dir}/dist/otomi.js --"
+
+stack_volume=""
+tmp_volume_dir=$(mktemp -d)
+tmp_volume=" -v $tmp_volume_dir:/tmp"
+if [[ $base_dir == *"otomi-core"* ]] || [[ $(pwd) == *"otomi-core"* ]]; then
+  silent echo "Checking if in dev environment"
+  if [[ $base_dir == *"otomi-core"* ]]; then
+    stack_dir=$(cd "$base_dir/.." && pwd)
+  elif [[ $(pwd) == *"otomi-core"* ]]; then
+    stack_dir=$(pwd | awk -F'otomi-core' '{print $1"otomi-core"}')
+  fi
+
+  stack_volume="-v $stack_dir:$stack_dir"
+  tmp_volume="-v /tmp:/tmp"
+
+  # Use development distribution if exists
+  if [ -f "${stack_dir}/dist/otomi.js" ]; then
+    executable="node --experimental-specifier-resolution=node ${stack_dir}/dist/otomi.js --"
+  else
+    executable="node --no-warnings --experimental-specifier-resolution=node --loader ts-node/esm ${stack_dir}/src/otomi.ts --"
+  fi
+  echo "OTOMI_DEV=true" >>$tmp_env
+fi
+
+check_volume_path() {
+  source=$1
+  dest=$2
+  [ -z $source ] && [ -z $dest ] && echo "" && return 0
+  echo "-v $source:$dest"
+  return 0
+}
+
+cmd="${executable} $*"
+# If command is "otomi bash"
+if [ "$1" = "bash" ] && [ "$#" = "1" ]; then
+  cmd="bash"
+fi
+
+if [ "$OTOMI_TAG" = 'latest' ] || [ "$OTOMI_TAG" = 'master' ] && [ -z $NOPULL ]; then 
+  silent echo "Pulling latest version of the docker image, please wait"
+  silent docker pull $otomi_tools_image
+  status=$?
+  if [ "$status" -ne 0 ]; then
+    echo "Something went wrong when trying to pull '${otomi_tools_image}'"
+    exit $status
+  fi
+fi
+
+it='-it'
+if [[ $calling_args == *'-no-interactive'* ]] ||
+    [[ $calling_args == *'-ni'* ]]; then
+  it=''
+fi
+
+network='--network host'
```

Comment:
Why does otomi need host network?

Reply:
According to @Morriz  it has to do with Google stuff, if I remember correctly. Not 100% sure, but there was a reason